dart/lib/src/protocol/sentry_event.dart

```diff
@@ -316,18 +320,24 @@ class SentryEvent {
           breadcrumbs.map((b) => b.toJson()).toList(growable: false);
     }
 
-    json['sdk'] = sdk?.toJson() ??
-        <String, String>{
-          'name': sdkName,
-          'version': sdkVersion,
-        };
+    Map<String, dynamic> sdkMap;
+    if (sdk != null && (sdkMap = sdk.toJson()).isNotEmpty) {
+      json['sdk'] = sdkMap;
+    } else {
+      json['sdk'] = <String, String>{
+        'name': sdkName,
+        'version': sdkVersion,
+      };
```

Comment:
I'd remove this just for the sake of DRY, `_prepareEvent` already sets it.

Reply:
✅